chrome/browser/ui/views/ime_driver/input_method_bridge_linux.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CHROME_BROWSER_UI_VIEWS_IME_DRIVER_INPUT_METHOD_BRIDGE_LINUX_H_
+#define CHROME_BROWSER_UI_VIEWS_IME_DRIVER_INPUT_METHOD_BRIDGE_LINUX_H_
+
+#include "chrome/browser/ui/views/ime_driver/remote_text_input_client.h"
+#include "services/ui/public/interfaces/ime/ime.mojom.h"
+
+namespace ui {
+class InputMethodAuraLinux;
+}
+
+// This bridges between mojo InputMethod API and aura::InputMethodBridgeLinux.
+// It forwards the received events to an instance of
+// aura::InputMethodBridgeLinux.
+class InputMethodBridgeLinux : public ui::mojom::InputMethod {
```

Comment:
Very likely this could be moved to ui/. But for that, RemoteTextInputClient also needs to be moved to ui/. Maybe to ui/base/ime?

I've been checking and actually all IMEDriver could be moved to ui.

Reply:
Agree but I don't think it's a good idea to do it in this commit

Reply:
Agree. Also, very likely this refactor could be landed upstream earlier than the desktop wayland specific changes. So that refactor could be directly contributed upstream.